ogr/ogrsf_frmts/miramon/ogrmiramondriver.cpp

```diff
@@ -0,0 +1,249 @@
+                 poOpenInfo->pabyHeader[4] == '1' &&
+                 poOpenInfo->pabyHeader[5] == '.' &&
+                 poOpenInfo->pabyHeader[6] == '1') ||
+                (poOpenInfo->pabyHeader[3] == ' ' &&
+                 poOpenInfo->pabyHeader[4] == '2' &&
+                 poOpenInfo->pabyHeader[5] == '.' &&
+                 poOpenInfo->pabyHeader[6] == '0'))
+            {
+                return TRUE;
+            }
+        }
+    }
+
+    return FALSE;
+}
+
+/****************************************************************************/
+/*                           OGRMiraMonDriverOpen()                         */
+/****************************************************************************/
+
+static GDALDataset *OGRMiraMonDriverOpen(GDALOpenInfo *poOpenInfo)
+
+{
+    if (!poOpenInfo->bStatOK)
+        return nullptr;
+
+    OGRMiraMonDataSource *poDS = new OGRMiraMonDataSource();
+
+    if (poDS != nullptr && poOpenInfo->eAccess == GA_Update)
+    {
+        CPLError(CE_Failure, CPLE_OpenFailed,
+                 "MiraMonVector driver does not support update.");
+        delete poDS;
+        poDS = nullptr;
+    }
+    else
+    {
+        if (!poDS->Open(poOpenInfo->pszFilename, nullptr, nullptr,
+                        poOpenInfo->eAccess == GA_Update,
+                        poOpenInfo->papszOpenOptions))
+        {
+            delete poDS;
+            poDS = nullptr;
+        }
+    }
+
+    return poDS;
+}
+
+/****************************************************************************/
+/*                         OGRMiraMonDriverCreate()                              */
+/****************************************************************************/
+
+static GDALDataset *
+OGRMiraMonDriverCreate(const char *pszName, CPL_UNUSED int /*nBands*/,
+                       CPL_UNUSED int /*nXSize*/, CPL_UNUSED int /*nYSize*/,
+                       CPL_UNUSED GDALDataType /*eDT*/, char **papszOptions)
+{
+    OGRMiraMonDataSource *poDS = new OGRMiraMonDataSource();
+
+    if (poDS->Create(pszName, papszOptions))
+        return poDS;
+
+    delete poDS;
+    return nullptr;
+}
+
+/****************************************************************************/
+/*                           RegisterOGRMM()                                */
+/****************************************************************************/
+
+void RegisterOGRMiraMon()
+
+{
+    if (GDALGetDriverByName("MiraMonVector") != nullptr)
+        return;
+
+    GDALDriver *poDriver = new GDALDriver();
+    poDriver->SetDescription("MiraMonVector");
+    poDriver->SetMetadataItem(GDAL_DCAP_VECTOR, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_CREATE_LAYER, "YES");
+    poDriver->SetMetadataItem(GDAL_DCAP_CREATE_FIELD, "YES");
+    poDriver->SetMetadataItem(GDAL_DMD_LONGNAME,
+                              "MiraMon Vectors (.pol, .arc, .pnt)");
+    poDriver->SetMetadataItem(GDAL_DMD_EXTENSION, "pol");
```

Comment:
GDAL_DMD_EXTENSION is legacy. And it doesn't make sense to define both EXTENSION singular and plural, son only keep EXTENSIONS plural
```suggestion
```

Reply:
Good to have you to explain that kind of things. Deleted